edx_exams/apps/api/v1/views.py

```diff
@@ -1 +1,171 @@
+                        )
+                        log.info(
+                            "Marked exam=%(previous_exam_id)s as inactive, created new exam=%(current_exam_id)s",
+                            {
+                                'previous_exam_id': exam_object.id,
+                                'current_exam_id': current.id,
+                            }
+                        )
+                    else:
+                        # if exam type has not changed, update the fields of the existing exam
+                        exam_object.exam_name = exam['exam_name']
+                        exam_object.time_limit_mins = exam['time_limit_mins']
+                        exam_object.due_date = exam['due_date']
+                        exam_object.hide_after_due = exam['hide_after_due']
+                        exam_object.is_active = exam['is_active']
+                        exam_object.save()
+                        log.info(
+                            "Updated existing exam=%(current_exam_id)s",
+                            {
+                                'current_exam_id': exam_object.id,
+                            }
+                        )
+
+                else:
+                    # if exam does not already exist, create a new one
+
+                    # check that configuration exists
+                    config = CourseExamConfiguration.objects.filter(course_id=exam['course_id'])
+
+                    if len(config) > 0:
+                        provider = config[0].provider
+
+                    else:
+                        # if configuration doesn't already request, return 400
+                        error_msg = "Cannot update exam. No exam configuration set up for course"
+                        return Response(error_msg, status=status.HTTP_400_BAD_REQUEST)
+
+                    current = Exam.objects.create(
+                        resource_id=str(uuid.uuid4()),
+                        course_id=exam['course_id'],
+                        provider=provider,
+                        content_id=exam['content_id'],
+                        exam_name=exam['exam_name'],
+                        exam_type=exam['exam_type'],
+                        time_limit_mins=exam['time_limit_mins'],
+                        due_date=exam['due_date'],
+                        hide_after_due=exam['hide_after_due'],
+                        is_active=exam['is_active']
+                    )
+                    log.info(
+                        "Created new exam=%(current_exam_id)s",
+                        {
+                            'current_exam_id': current.id,
+                        }
+                    )
+
+            # mark any exams not included in the request as inactive. The Query set has already been filtered by course
+            remaining_exams = course_exams.exclude(content_id__in=visited_exams)
+            remaining_exams.update(is_active=False)
```

Comment:
I think this might mark any new exams and is_active = false
changed exams are ok even though the exam object is new because they have the same content ID

Reply:
It shouldn't, because any content ID (whether it's new or existing) is added to visited_exams

Reply:
yeah I was just reading again and realized visited is in the way outer serializer loop so my original comment was just wrong. Although maybe fuel to your feeling that there is a bit too much in this function.